### PR TITLE
Added k3 qmk keyboard

### PR DESCRIPTION
**Pull request by proceee**
Added  darmoshark k3 qmk keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/darmoshark/k3/.gitignore
Comment (zvecr) [inline keyboards/darmoshark/k3/.gitignore:1]: This file should be removed.

File: keyboards/darmoshark/k3/info.json [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    },
+    "ws2812": {
+        "pin": "A8"
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
+        "center_point": [24, 32],
+        "max_brightness": 140,
+        "val_steps": 35,
+        "speed_steps": 64,
```
Comment (zvecr) [inline keyboards/darmoshark/k3/info.json:40]:
```suggestion
```

File: keyboards/darmoshark/k3/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ),
+
+    [3] = LAYOUT(
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______,
+        _______, _______, _______, _______,
+        _______,          _______
+    )
```
Comment (zvecr) [inline keyboards/darmoshark/k3/keymaps/default/keymap.c:42]:
```suggestion
    )
```

File: keyboards/darmoshark/k3/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    ),
+
+    [3] = LAYOUT(
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______,
+        _______, _______, _______, _______,
+        _______,          _______
+    )
```
Comment (zvecr) [inline keyboards/darmoshark/k3/keymaps/via/keymap.c:42]:
```suggestion
    )
```

File: keyboards/darmoshark/k3/readme.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+
+# K3 QMK
```
Comment (zvecr) [inline keyboards/darmoshark/k3/readme.md:2]:
```suggestion
# K3 QMK
```

File: keyboards/darmoshark/k3/readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+* Hardware Supported: [Darmoshark](http://www.Darmoshark.cn)
+* Hardware Availability: [Darmoshark](http://www.Darmoshark.cn)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make k3:default
+
+Flashing example for this keyboard:
+
+    make k3:default:flash
```
Comment (zvecr) [inline keyboards/darmoshark/k3/readme.md:14]:
```suggestion
    make darmoshark/k3:default

Flashing example for this keyboard:

    make darmoshark/k3:default:flash
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/darmoshark/k3/mcuconf.h [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2023 Proceee
```
Comment (tzarc) [inline keyboards/darmoshark/k3/mcuconf.h:1]: Remove this file, it is not used.

File: keyboards/darmoshark/k3/info.json [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+                { "matrix":[3,1], "x": 1, "y": 3 },
+                { "matrix":[3,2], "x": 2, "y": 3 },
+
+                { "matrix":[4,0], "x": 0, "y": 4 },
+                { "matrix":[4,1], "x": 1, "y": 4 },
+                { "matrix":[4,2], "x": 2, "y": 4 },
+                { "matrix":[4,3], "x": 3, "y": 4, "h": 2 },
+
+                { "matrix":[5,1],  "x": 0, "y": 5, "w": 2 },
+                { "matrix":[5,2],  "x": 2, "y": 5 }
```
Comment (zvecr) [inline keyboards/darmoshark/k3/info.json:167]:
```suggestion
                { "matrix":[2,2], "x": 2, "y": 2 },

                { "matrix":[3,0], "x": 0, "y": 3 },
                { "matrix":[3,1], "x": 1, "y": 3 },
                { "matrix":[3,2], "x": 2, "y": 3 },
                { "matrix":[2,3], "x": 3, "y": 2, "h": 2 },

                { "matrix":[4,0], "x": 0, "y": 4 },
                { "matrix":[4,1], "x": 1, "y": 4 },
                { "matrix":[4,2], "x": 2, "y": 4 },

                { "matrix":[5,1],  "x": 0, "y": 5, "w": 2 },
                { "matrix":[5,2],  "x": 2, "y": 5 },
                { "matrix":[4,3], "x": 3, "y": 4, "h": 2 }
```

File: keyboards/darmoshark/k3/info.json [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+            { "flags": 4, "x": 224, "y": 64},
+            { "flags": 4, "x": 224, "y": 64},
+            { "flags": 4, "x": 224, "y": 64},
+            { "flags": 4, "x": 224, "y": 64},
+            { "flags": 4, "x": 224, "y": 64},
+            { "flags": 4, "x": 224, "y": 64}
+        ]
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/darmoshark/k3/info.json:140]:
```suggestion
    "community_layouts": ["numpad_6x4"],
    "layouts": {
        "LAYOUT_numpad_6x4": {
```

**Comment by tzarc** [comment]
Also, it would be much appreciated if you stopped changing github handles every few boards.

File: keyboards/darmoshark/k3/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        KC_P0,              KC_PDOT
+    ),
+
+    [1] = LAYOUT(
+        EE_CLR,   _______,  RGB_MOD,  _______,
+        KC_CALC,  _______,  _______,  RGB_VAD,
+        _______,  RGB_SPI,  _______,  RGB_VAI,
+        _______,  _______,  _______,
+        _______,  RGB_SPD,  _______,  _______,
+        _______,            RGB_TOG
```
Comment (zvecr) [inline keyboards/darmoshark/k3/keymaps/default/keymap.c:23]:
```suggestion
    [0] = LAYOUT_numpad_6x4(
        KC_ESC,   KC_TAB,   KC_BSPC,  MO(1),
        KC_NUM,   KC_PSLS,  KC_PAST,  KC_PMNS,
        KC_P7,    KC_P8,    KC_P9, 
        KC_P4,    KC_P5,    KC_P6, KC_PPLS,
        KC_P1,    KC_P2,    KC_P3,
        KC_P0,              KC_PDOT,    KC_PENT
    ),

    [1] = LAYOUT_numpad_6x4(
        EE_CLR,   _______,  RGB_MOD,  _______,
        KC_CALC,  _______,  _______,  RGB_VAD,
        _______,  RGB_SPI,  _______,
        _______,  _______,  _______, RGB_VAI,
        _______,  RGB_SPD,  _______, 
        _______,            RGB_TOG,    _______
```

File: keyboards/darmoshark/k3/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        KC_P0,              KC_PDOT
+    ),
+
+    [1] = LAYOUT(
+        EE_CLR,   _______,  RGB_MOD,  _______,
+        KC_CALC,  _______,  _______,  RGB_VAD,
+        _______,  RGB_SPI,  _______,  RGB_VAI,
+        _______,  _______,  _______,
+        _______,  RGB_SPD,  _______,  _______,
+        _______,            RGB_TOG
```
Comment (zvecr) [inline keyboards/darmoshark/k3/keymaps/via/keymap.c:23]:
```suggestion
    [0] = LAYOUT_numpad_6x4(
        KC_ESC,   KC_TAB,   KC_BSPC,  MO(1),
        KC_NUM,   KC_PSLS,  KC_PAST,  KC_PMNS,
        KC_P7,    KC_P8,    KC_P9, 
        KC_P4,    KC_P5,    KC_P6, KC_PPLS,
        KC_P1,    KC_P2,    KC_P3,
        KC_P0,              KC_PDOT,    KC_PENT
    ),

    [1] = LAYOUT_numpad_6x4(
        EE_CLR,   _______,  RGB_MOD,  _______,
        KC_CALC,  _______,  _______,  RGB_VAD,
        _______,  RGB_SPI,  _______,
        _______,  _______,  _______, RGB_VAI,
        _______,  RGB_SPD,  _______, 
        _______,            RGB_TOG,    _______
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/darmoshark/k3/mcuconf.h (hunk 6 above)
Comment (tzarc) [reply keyboards/darmoshark/k3/mcuconf.h:1]: Ignore this, you're using the QSPI peripheral in SPI mode.

**Comment by proceee** [comment]
> Also, it would be much appreciated if you stopped changing github handles every few boards.

I don't quite see your point

File: keyboards/darmoshark/k3/info.json [hunk 11]
```diff
@@ -0,0 +1,172 @@
+{
+    "manufacturer": "Darmoshark",
+    "keyboard_name": "K3 QMK",
+    "maintainer": "Proceee",
+    "url": "",
+    "processor": "WB32FQ95",
+    "bootloader": "wb32-dfu",
+    "usb": {
+        "device_version": "1.0.0",
+        "vid": "0x3555",
```
Comment (drashna) [inline keyboards/darmoshark/k3/info.json:10]: https://www.the-sz.com/products/usbid/index.php?v=0x3555&p=&n=

File: keyboards/darmoshark/k3/rules.mk [hunk 12]
```diff
@@ -0,0 +1,2 @@
+EEPROM_DRIVER = wear_leveling
```
Comment (dunk2k) [inline keyboards/darmoshark/k3/rules.mk:1]: migrate to `info.json`
```suggestion
```

File: keyboards/darmoshark/k3/info.json [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+    "diode_direction": "ROW2COL",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
```
Comment (dunk2k) [inline keyboards/darmoshark/k3/info.json:23]:
```suggestion
    },
    "eeprom": {
        "driver": "wear_leveling"
    },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/darmoshark/k3/info.json [hunk 14]
```diff
@@ -0,0 +1,175 @@
+{
+    "manufacturer": "Darmoshark",
+    "keyboard_name": "K3 QMK",
+    "maintainer": "Proceee",
+    "url": "",
+    "processor": "WB32FQ95",
+    "bootloader": "wb32-dfu",
+    "usb": {
+        "device_version": "1.0.0",
+        "vid": "0x3555",
```
Comment (proceee) [inline keyboards/darmoshark/k3/info.json:10]:
```suggestion
        "vid": "0xC001",
```

**Review by drashna — APPROVED** [review]

**Comment by proceee** [comment]
Can merge？

File: keyboards/darmoshark/k3/config.h [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
+/* RGB Matrix */
+#define RGB_MATRIX_LED_COUNT 44
+#define RGB_DISABLE_WHEN_USB_SUSPENDED
+#define RGB_MATRIX_DEFAULT_VAL 80
+
```
Comment (zvecr) [inline keyboards/darmoshark/k3/config.h:15]:
```suggestion

```

File: keyboards/darmoshark/k3/config.h [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+#define SPI_SCK_PIN B3
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+#define SPI_MOSI_PAL_MODE 5
+
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
+#define WEAR_LEVELING_BACKING_SIZE (4 * 1024)
+
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
```
Comment (zvecr) [inline keyboards/darmoshark/k3/config.h:27]:
```suggestion

#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
```

File: keyboards/darmoshark/k3/rules.mk [hunk 17]
```diff
@@ -0,0 +1 @@
+WEAR_LEVELING_DRIVER = spi_flash
```
Comment (zvecr) [inline keyboards/darmoshark/k3/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

File: keyboards/darmoshark/k3/info.json [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "eeprom": {
+        "driver": "wear_leveling"
```
Comment (zvecr) [inline keyboards/darmoshark/k3/info.json:25]:
```suggestion
        "driver": "wear_leveling",
        "wear_leveling": {
            "driver": "spi_flash",
            "backing_size": 4096
        }
```

File: keyboards/darmoshark/k3/info.json [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+    },
+    "indicators": {
+        "num_lock": "C5",
+        "on_state": 1
+    },
+    "ws2812": {
+        "pin": "A8"
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
```
Comment (zvecr) [inline keyboards/darmoshark/k3/info.json:39]:
```suggestion
        "driver": "ws2812",
        "default":{
          "val": 80
        },
        "sleep": true.
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/darmoshark/k3/info.json [hunk 20]
```diff
@@ -0,0 +1,183 @@
...
+        "default":{
+          "val": 80
+        },
+        "sleep": true.
```
Comment (zvecr) [inline keyboards/darmoshark/k3/info.json:47]: Sorry, fixes previous bad suggesion 
```suggestion
        "sleep": true,
```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**